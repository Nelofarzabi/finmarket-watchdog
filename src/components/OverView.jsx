import React from 'react';

const OverView = () => (
  <div>OverView</div>
);

export default OverView;
