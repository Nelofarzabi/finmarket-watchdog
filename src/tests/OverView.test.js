import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { useSelector } from 'react-redux';
import OverView from '../components/OverView';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('OverView', () => {
  const mockCompanyDetail = {
    image: 'image-url.jpg',
  };

  beforeEach(() => {
    useSelector.mockImplementation((selector) => selector({
      companies: {
        companyDetail: mockCompanyDetail,
      },
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders the OverView component with company image', () => {
    const { getByText, getByAltText } = render(
      <MemoryRouter>
        <OverView />
      </MemoryRouter>,
    );

    // Check if the component renders the "Back Home" link
    expect(getByText('Back Home')).toBeInTheDocument();

    // Check if the component renders the company image
    const companyImage = getByAltText('Company');
    expect(companyImage).toBeInTheDocument();
    expect(companyImage.src).toContain('image-url.jpg');

    // Check if the component renders the navigation links
    expect(getByText('Info')).toBeInTheDocument();
    expect(getByText('Description')).toBeInTheDocument();
    expect(getByText('Statement')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <OverView />
      </MemoryRouter>,
    );

    // Check if the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });
});
