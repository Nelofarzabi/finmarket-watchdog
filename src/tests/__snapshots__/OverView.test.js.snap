// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverView matches snapshot 1`] = `
<div>
  <div
    class="mb-3 px-4"
  >
    <a
      class="flex gap-4 items-center mb-4"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="text-white text-2xl"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <p
        class="text-white text-xl"
      >
        Back Home
      </p>
    </a>
    <div
      class="flex justify-center items-start gap-6"
    >
      <div
        class=""
      >
        <img
          alt="Company"
          class=" object-cover"
          src="image-url.jpg"
        />
      </div>
    </div>
    <nav
      class="py-2 w-full "
    >
      <ul
        class="flex md:w-[55%] mx-auto justify-between gap-4"
      >
        <li>
          <a
            aria-current="page"
            class="text-white text-xl active"
            href="/"
            style="border-bottom: 1px solid #fff; color: rgb(244, 238, 224); padding: 2px;"
          >
            Info
          </a>
        </li>
        <li>
          <a
            class="text-white text-xl"
            href="/description"
          >
            Description
          </a>
        </li>
        <li>
          <a
            class="text-white text-xl"
            href="/statement"
          >
            Statement
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
